src/main/java/TaskList.java

```diff
@@ -0,0 +1,59 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents a List of tasks that the user wants to keep track of and methods to interact with the list.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    public TaskList(File file, Ui ui) {
+        this.tasks = new ArrayList<>();
+
+        try {
+            Scanner sc = new Scanner(file);
+
+            while (sc.hasNextLine()) {
+                String[] components = sc.nextLine().split(" \\| ");
+
+                switch (components[0]) {
+                    case "T":
+                        tasks.add(new Todo(components[2], components[1].equals("1") ? true : false));
+                        break;
+                    case "D":
+                        tasks.add(new Deadline(components[2], components[3], components[1].equals("1") ? true : false));
+                        break;
+                    case "E":
+                        tasks.add(new Event(components[2], components[3], components[1].equals("1") ? true : false));
+                        break;
+                }
+            }
```

Comment:
Perhaps you can split and name the components right from the start for 1, 2 and 3 to make the code more easily understood.